### DISCUSSION TITLE
Debian 12 bookworm and nftables won't block

### DISCUSSION THREAD

**Discussion by jameskimmel** [fail2ban/fail2ban#3575] [Q&A]
Hi guys

I am fairly new to fail2ban but unfortunately was unable to setup some rules for a jellyfin host on debian 12. 
The IP is banned, but I am still able to connect to the page. I suspect the switch from iptables to nftables to be the problem. 

```
sudo nano /etc/fail2ban/jail.d/defaults-debian.conf
[sshd]
backend=systemd
enabled = true

[DEFAULT]
banaction = nftables
banaction_allports = nftables[type=allports]

```

Setting the backend was needed to even get fail2ban to start. Setting the banaction is how I think the switch to nftables is done?

```
sudo fail2ban-client status jellyfin
Status for the jail: jellyfin
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     0
|  `- Journal matches:
`- Actions
   |- Currently banned: 1
   |- Total banned:     1
   `- Banned IP list:   10.0.51.3

```

But that banned IP is still able to connect. Any ideas? Thanks!

**Comment by sebres** [comment]
Basically there may be too many reasons to say something without details: jail config (`fail2ban-client -d | grep jellyfin`), output of nft chains and rules, etc.

Also note the [FAQ in the wiki](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works) (answer to 2nd question)...

**Reply by jameskimmel** [reply]
Hey sebres, thank you for your time. 

Well, there was two mistakes that I made.
First one was to assume that I don't have to enable nftables. But it seems like nftables is not enabled by default, so unless you enable it manually, fail2ban does nothing to block anything. Maybe there could be a warning for users or some checks during the installation? 

Am I right to assume, that by default fail2ban uses iptables, which is why I have to manually change the banaction? If so, could this not be the default for defaults-debian.conf?

The second mistake was to test if I can reach the page, by inserting the password three times, which blocked my internal IPv4. What I did not realise is that fail2ban actually did block my IPv4, but that after that my browser just switched to IPv6 to reach the host :)

**Reply by jameskimmel** [reply]
Sorry to reopen this, is there a reason why we should not change the default of defaults-debian.conf to this?

```
sudo nano /etc/fail2ban/jail.d/defaults-debian.conf
[sshd]
backend=systemd
enabled = true

[DEFAULT]
banaction = nftables
banaction_allports = nftables[type=allports]

```

**Reply by sebres** [reply]
Regarding defaults - basically this is a matter of maintainers for Debian distribution.
Also `jail.d/defaults-debian.conf` is not a part of this repository - it is invention of them and part of [salsa.debian.org/.../packages/fail2ban](https://salsa.debian.org/python-team/packages/fail2ban/-/blob/master/debian/debian-files/jail.d_defaults-debian.conf).
And I don't really like it: neither I believe the idea to enable jails by default is good, nor I think this is appropriate place for that, especially because the config for jail(s) belongs either to `jail.local` (preferred single file configuration) or to `jail.d/<jail>.conf`. But also (if we talking about `backend = systemd` for some jails) because we already have distribution-related includes, like this: https://github.com/fail2ban/fail2ban/blob/f8f8c046a277fbf17fcc0a3075234adb07274dcf/config/jail.conf#L36

Anyway this configuration is pretty simple and everyone can do that by himself.
You can discuss it with @fail2ban/maintainers (e. g. with @sylvestre).
Or if you'd like, you can add a new wiki page or extend some existing like [wiki :: Proper fail2ban configuration](https://github.com/fail2ban/fail2ban/wiki/Proper-fail2ban-configuration)... 

However then it should rather look like this (it belongs to `jail.local` and default section first):
```
# sudo nano /etc/fail2ban/jail.local

[DEFAULT]
banaction = nftables
banaction_allports = nftables[type=allports]

[sshd]
backend=systemd
enabled = true
```

**Reply by sylvestre** [reply]
happy to make the change in Debian

**Reply by sylvestre** [reply]
Done:
https://salsa.debian.org/python-team/packages/fail2ban/-/commit/bc37c9413716b0f238bb154f00094a27a0670284

I will upload it in the next few days

**Reply by sebres** [reply]
Hmm... I'm unsure this default section would affect any jail excepting sshd specified here...
IIRC `jail.d/*.conf` are intended to load single jail (at least if it were `jail.d/sshd.conf`, unsure about different names) and then must affect only defaults related to that single jail. 
But also if it accidentally works (so affects every jail no matter where it is specified), it looks ambiguous to me, but still worse we can't guarantee the load mechanisms would not change later and it would only affect jails specified in that file.
In my opinion better and more consistent would be to have `/etc/fail2ban/defaults-debian.conf` and something like that in `jail.conf`:
```diff
  [INCLUDES]
  before = paths-debian.conf
+ after = defaults-debian.conf
```

**Reply by jameskimmel** [reply]
Well, you guys are way more competent than me to decide how to solve this in the most elegant way :)
Do you wanna discuss it further or should I close this discussion?

**Comment by JuergenKosel** [comment]
I have set in `/etc/fail2ban/jail.local` :+1: 

```conf
[DEFAULT]
banaction = nftables-multiport
```

And this solved the problem for me.
Taken from https://blog.cyberfront.org/index.php/2021/10/27/debian-fail2ban/

**Reply by eabase** [reply]
What happened to: `banaction_allports` ?
